src/combinatorics/simple_graph/basic.lean

```diff
@@ -94,39 +97,119 @@ by { classical, exact fintype.of_injective simple_graph.adj simple_graph.ext }
 
 @[simp]
 lemma simple_graph.from_rel_adj {V : Type u} (r : V → V → Prop) (v w : V) :
-  (simple_graph.from_rel r).adj v w ↔ v ≠ w ∧ (r v w ∨ r w v) :=
-iff.rfl
+  (simple_graph.from_rel r).adj v w ↔ v ≠ w ∧ (r v w ∨ r w v) := iff.rfl
 
-/--
-The complete graph on a type `V` is the simple graph with all pairs of distinct vertices adjacent.
--/
-def complete_graph (V : Type u) : simple_graph V :=
-{ adj := ne }
+/-- The complete graph on a type `V` is the simple graph with all pairs of distinct vertices
+adjacent. In `mathlib`, this is usually referred to as `⊤`. -/
+def complete_graph (V : Type u) : simple_graph V := { adj := ne }
 
-instance (V : Type u) : inhabited (simple_graph V) :=
-⟨complete_graph V⟩
-
-instance complete_graph_adj_decidable (V : Type u) [decidable_eq V] :
-  decidable_rel (complete_graph V).adj :=
-λ v w, not.decidable
-
-/-- The graph with no edges on a given vertex type `V`. -/
-def empty_graph (V : Type u) : simple_graph V :=
-{ adj := λ i j, false }
+/-- The graph with no edges on a given vertex type `V`. `mathlib` prefers the notation `⊥`. -/
+def empty_graph (V : Type u) : simple_graph V := { adj := λ i j, false }
 
 namespace simple_graph
 
 variables {V : Type u} {W : Type v} {X : Type w} (G : simple_graph V) (G' : simple_graph W)
 
+@[simp] lemma irrefl {v : V} : ¬G.adj v v := G.loopless v
+
+lemma adj_comm (u v : V) : G.adj u v ↔ G.adj v u := ⟨λ x, G.sym x, λ x, G.sym x⟩
+
+@[symm] lemma adj_symm {u v : V} (h : G.adj u v) : G.adj v u := G.sym h
+
+lemma ne_of_adj {a b : V} (hab : G.adj a b) : a ≠ b :=
+by { rintro rfl, exact G.irrefl hab }
+
+section order
+
+/-- The relation that one `simple_graph` is a subgraph of another. -/
+def is_subgraph (x y : simple_graph V) : Prop := ∀ ⦃v w : V⦄, x.adj v w → y.adj v w
+
+instance : boolean_algebra (simple_graph V) :=
+{ le := is_subgraph,
+  sup := λ x y,
+        { adj := x.adj ⊔ y.adj,
+          sym := λ v w h, by rwa [sup_apply, sup_apply, x.adj_comm, y.adj_comm] },
+  inf := λ x y,
+        { adj := x.adj ⊓ y.adj,
+          sym := λ v w h, by rwa [inf_apply, inf_apply, x.adj_comm, y.adj_comm] },
+  compl := λ G,
+        { adj := λ v w, v ≠ w ∧ ¬G.adj v w,
+          sym := λ v w ⟨hne, _⟩, ⟨hne.symm, by rwa adj_comm⟩,
+          loopless := λ v ⟨hne, _⟩, (hne rfl).elim },
+  sdiff := λ x y,
+        { adj := x.adj \ y.adj,
+        sym := λ v w h, by change x.adj w v ∧ ¬ y.adj w v; rwa [x.adj_comm, y.adj_comm] },
```

Comment:
To be clear, I think it's _maybe_ still worth having separate `has_sup`, `has_inf` etc instances for these, because it gives you a nice place to put a docstring for each.

Reply:
I tried that, but then wasn't sure how to include them; I didn't like `simple_graph.has_compl.1`, and also I can use `add_doc_comment`. They're pretty self-explanatory instances, to be honest

Reply:
You can include them like this:
```suggestion
  sup := (⊔),
  inf := (⊓)
  compl := has_compl.compl,
  sdiff := (\),
```